activation/interface.go

```diff
@@ -29,12 +29,21 @@ type scaler interface {
 	scale(int)
 }
 
+type nipostChallenge interface {
+	Hash() types.Hash32
+	Publish() types.EpochID
+	PrevATX() types.ATXID
+	PositioningATX() types.ATXID
+	CommitmentATX() *types.ATXID
+	MaybeSequence() *uint64
+}
+
 // validatorOption is a functional option type for the validator.
 type validatorOption func(*validatorOptions)
 
 type nipostValidator interface {
-	InitialNIPostChallenge(challenge *types.NIPostChallenge, atxs atxProvider, goldenATXID types.ATXID) error
-	NIPostChallenge(challenge *types.NIPostChallenge, atxs atxProvider, nodeID types.NodeID) error
+	InitialNIPostChallenge(challenge nipostChallenge, atxs atxProvider, goldenATXID types.ATXID) error
+	NIPostChallenge(challenge nipostChallenge, atxs atxProvider, nodeID types.NodeID) error
```

Comment:
Why the interface? It is possible to just pass a `*wire.NIPostChallengeV1` there is no cyclic dependency; the validator isn't used outside of the activation package or packages already importing the `activation` package.

This would also get rid of the need for implementing "dummy" methods on the `*wire.NIPostChallengeV1` type just to satisfy this interface.